lib/resolve-decl.js

```diff
@@ -41,29 +119,107 @@ function eachMapItemDependencyOfDecl(variablesUsedList, map, decl, cb) {
 				//console.log(generateScopeList(mapItem.parent, true));
 				//console.log('amd isNodeUnderScope', isNodeUnderScope(mimicDecl.parent, mapItem.parent), mapItem.decl.value);
 			}
-			// TODO: use regex from `isUnderScope`
-			else if(isUnderScope.RE_PSEUDO_SELECTOR.test(mapItem.parent.selector)) {
-				// Create a detached clone
-				var ruleClone = shallowCloneNode(decl.parent);
-				ruleClone.parent = decl.parent.parent;
-
-				// Add the declaration to it
-				mimicDecl = decl.clone();
-				ruleClone.append(mimicDecl);
-
-				var lastPseudoSelectorMatches = mapItem.parent.selector.match(new RegExp(isUnderScope.RE_PSEUDO_SELECTOR.source + '$'));
-				var lastPseudoSelector = lastPseudoSelectorMatches ? lastPseudoSelectorMatches[2] : '';
-
-				ruleClone.selector += lastPseudoSelector;
+			else if(mapItem.parent.selector !== decl.parent.selector && declSelector) {
+				var s = parseSelector(mapItem.parent);
+
+				log(s);
+
+				var append = null;
+				var idx = 0;
+				var process = s.length > 1;
+				for(var i = 0; i < declSelector.length; i++) {
+					var a = declSelector[i];
+					var b = findIndex(s, c => compatible(c, a), idx);
+					process |= s.findIndex(c => compatible2(c, a), idx) != -1;
+
+					if(b < idx) {
+						// no match: already compatible or wildcard
+						if(i === 0) {
+							state = 1;
```

Comment:
Can these states be nice string label constants or symbols? Reasoning about these arbitrary state numbers seems mind bending 🤯